config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +106,232 @@ const (
 	// UserDefinedNamespaceIsolated enables monitoring for user-defined projects with namespace-scoped tenancy. This ensures that metrics, alerts, and monitoring data are isolated at the namespace level.
 	UserDefinedNamespaceIsolated UserDefinedMode = "NamespaceIsolated"
 )
+
+// alertmanagerMainConfig provides configuration options for the default Alertmanager instance
+// that runs in the `openshift-monitoring` namespace. Use this configuration to control
+// whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
+//
+// +union
+type AlertmanagerConfig struct {
+	// deploymentMode determines whether the default Alertmanager instance should be deployed
+	// as part of the monitoring stack.
+	// Allowed values are Deployed and NotDeployed
+	// When set to Deployed, the Cluster Monitoring Operator
+	// ensures that an Alertmanager instance is created and managed in the `openshift-monitoring` namespace.
+	// When set to NotDeployed, the operator will not deploy the Alertmanager instance.
+	// Use this field if you want to explicitly opt in or out of running a platform-level Alertmanager.
+	//
+	// This filed is required
+	// Allowed values are Deployed and NotDeployed.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum=Deployed;NotDeployed
+	// +kubebuilder:validation:Required
+	DeploymentMode string `json:"deploymentMode"`
```

Comment:
The parent struct is optional, so what does it mean when the parent is omitted?

The parent also does not have omitempty, nor is it a pointer. Which means it is discoverable (++ for config API), however, this field being required, is going to cause issues. 

If I asked you to allow `""` as a valid value for the enum, what would that mean to the controller?